visa_mock/base/base_mocker.py

```diff
@@ -164,6 +164,11 @@ def __call__(self, mocker_self, *args, **kwargs):
         """
         new_args = ()
         new_kwargs = {}
+        annotation_types = [Union]
+        try:
+            annotation_types.append(types.UnionType)
+        except AttributeError:
+            pass
```

Comment:
This is a bit of a workaround for the fact that UnionType is unsupported in in Python version 3.8.  I think this condition needs to be clear, so that we can consolidate at a later time when we move fully to 3.10.

Consider using sys to check the python version explicitly and to act accordingly.  This will make the logic very clear to future readers of this code.

```suggestion
import sys

        # Check if types.UnionTypes is supported by current Python Version (3.10 or greater)
        annotation_types = [Union]
        if sys.version_info >= (3, 10):
              annotation_types.append(types.UnionType)
```

Reply:
I agree that this is cleaner, I tried it last night using `platform` and something similar to the one liner I've just made and it kept throwing errors for me. This seems to work though.

I've added some code comments to assist.